Tools/cases_generator/test_generator.py

```diff
@@ -4,6 +4,29 @@
 import tempfile
 
 import generate_cases
+from parser import StackEffect
```

Comment:
Question about this file: is it supposed to run as part of our test suite, or is the workflow just to use `pytest` or something locally?

Reply:
For now I just run it locally after I make changes.

I don't think our test suite supports pytest, and I don't want to make it a separate GitHub workflow. Eventually we'll have to rewrite it to use unittest. But not today.